### ISSUE TITLE
FieldAngle CLOCKWISE and OFFSET apply to all angle pickers simultaneously

### ISSUE THREAD

**Issue by rachel-fenichel** [google/blockly#2315]
### Problem statement
We allow angle pickers to work in both protractor and compass mode, but whatever mode is picked applies to all angle pickers on the screen.
```
/* The following two settings work together to set the behaviour of the angle
 * picker.  While many combinations are possible, two modes are typical:
 * Math mode.
 *   0 deg is right, 90 is up.  This is the style used by protractors.
 *   Blockly.FieldAngle.CLOCKWISE = false;
 *   Blockly.FieldAngle.OFFSET = 0;
 * Compass mode.
 *   0 deg is up, 90 is right.  This is the style used by maps.
 *   Blockly.FieldAngle.CLOCKWISE = true;
 *   Blockly.FieldAngle.OFFSET = 90;
 */

/**
 * Angle increases clockwise (true) or counterclockwise (false).
 */
Blockly.FieldAngle.CLOCKWISE = false;

/**
 * Offset the location of 0 degrees (and all angles) by a constant.
 * Usually either 0 (0 = right) or 90 (0 = up).
 */
Blockly.FieldAngle.OFFSET = 0;
```

### Expected Behavior

You can pick on a per-field basis.

### Additional Information
Blockly's angle picker doesn't have labels to help orient the user, so it might be confusing to allow these to mix.  But if we upgrade the rendering, we may also want to change these properties.

**Comment by abhinavsagar** [comment]
@rachel-fenichel Is this issue still open. Could I work on this?

**Comment by RoboErikG** [comment]
@abhinavsagar yep, should be fine for you to work on. As a note, this should be done by adding a property on the field to override the default behavior with a setter and a getter. The getter should handle checking the local property first and falling back to the global property if it's not set.

@alschmiedt do you have any thoughts on how this should be added to the JSON definition?

**Comment by alschmiedt** [comment]
Talked with Rachel and I think adding a 'mode' field to the JSON definition with the options for 'compass' and 'math' would be ideal. There aren't a lot of requests for other configurations, and most other configurations are pretty confusing anyway.

**Comment by abhinavsagar** [comment]
@alschmiedt @RoboErikG Got it i am taking on this work. I'll open a PR in a couple of days time.

**Comment by abhinavsagar** [comment]
Sorry for the late response. I am unable to solve this with my limited knowledge. Would look forward to solve other issues in this repository.

**Comment by rachel-fenichel** [comment]
No worries.  Feel free to grab a different issue instead.

**Comment by abhinavsagar** [comment]
Sure

**Comment by samelhusseini** [comment]
Fixed by #2654
